lib/Controller/ObjectsController.php

```diff
@@ -0,0 +1,359 @@
+            // Get all parameters from the request
+            $data = $this->request->getParams();
+            
+            // Get extend parameter if present
+            $extend = $data['extend'] ?? $data['_extend'] ?? [];
+            if (is_string($extend)) {
+                $extend = array_map('trim', explode(',', $extend));
+            }
+
+
+            // Remove the 'id' field if it exists, as we're creating a new object
+            unset($data['id']);
+
+            // Save the new object
+            $object = $this->objectService->saveObject(objectType: $objectType, object: $data, extend: $extend);
+            
+            // Return the created object as a JSON response
+            return new JSONResponse($object);
+        } catch (Exception $e) {
+            return new JSONResponse(
+                ['error' => $e->getMessage()],
+                400
+            );
+        }
+	}
+
+	/**
+	 * Update an object
+	 *
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 *
+	 * @return JSONResponse
+	 */
+	public function update(string $objectType, string $id): JSONResponse
+	{
+        try {
+            // Get all parameters from the request
+            $data = $this->request->getParams();
+            
+            // Get extend parameter if present
+            $extend = $data['extend'] ?? $data['_extend'] ?? [];
+            if (is_string($extend)) {
+                $extend = array_map('trim', explode(',', $extend));
+            }
+
+            // Ensure ID in data matches URL parameter
+            $data['id'] = $id;
+
+            // Save the updated object
+            $object = $this->objectService->saveObject(objectType: $objectType, object: $data, extend: $extend);
+            
+            // Return the updated object as a JSON response
+            return new JSONResponse($object);
+        } catch (Exception $e) {
+            return new JSONResponse(
+                ['error' => $e->getMessage()],
+                400
+            );
+        }
+	}
+
+	/**
+	 * Delete an object
+	 *
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 *
+	 * @return JSONResponse
+	 */
+	public function destroy(string $objectType, string $id): JSONResponse
+	{
+        try {
+            // Delete the object
+            $result = $this->objectService->deleteObject($objectType, $id);
+
+            // Return the result as a JSON response
+            return new JSONResponse(['success' => $result], $result === true ? 200 : 404);
+        } catch (Exception $e) {
+            return new JSONResponse(
+                ['error' => $e->getMessage()],
+                400
+            );
+        }
+	}
+
+	/**
+     * Get audit trail for a specific object
+     *
+     * @NoAdminRequired
+     * @NoCSRFRequired
+	 *
+	 * @return JSONResponse
+     */
+    public function getAuditTrail(string $objectType, string $id): JSONResponse
+    {
+        try {
+
+            $auditTrail = $this->objectService->getAuditTrail($objectType, $id);
```

Comment:
ObjectService does not have a function getAuditTrail()

Reply:
Still needs to be fixed